apps/demo-game/src/pages/play/cockpit.tsx

```diff
@@ -161,124 +160,229 @@ function Cockpit() {
+      val: `CHF ${assets.stocks.toFixed(2)}`,
     },
     {
       cat: 'Total',
-      val: `CHF ${resultFacts?.assets.totalAssets.toFixed(2)}`,
+      val: `CHF ${assets.totalAssets.toFixed(2)}`,
     },
   ]
 
-  const header = (
-    <div className="rounded border p-4">
-      <div className="font-bold">
-        Playing as {self?.name} in game {currentGame.id}
-      </div>
+  // TODO(JJ):
+  // - Level is not consistent with component
+  // Component wants a number, level is an object, though.
+  // console.log(self)
+  // - Do onclick logic
+  // - PlayerDisplay is not ideal/nice yet
+
+  const tabs = [
+    { name: 'Welcome', href: '/play/welcome' },
+    { name: 'Cockpit', href: '/play/cockpit' },
+  ]
+
+  const playerInfo = {
+    name: data.self.name,
+    color: data.self.facts.color,
+    location: data.self.facts.location,
+    level: data.self.level.id,
+    xp: data.self.experience,
+    xpMax: data.self.experienceToNext,
+    achievements: data.self.achievements,
+    imgPathAvatar: data.self.facts.avatar,
+    imgPathLocation: `/locations/${data.self.facts.location}.svg`,
+    onClick: () => {
+      // router.replace('/play/welcome')
+    },
+  }
+
+  // Transactions
+  // TODO(JJ): Integrate into the layout
+  // The following is only temporary
+  // - Should only display final transactions, or really every decision?
+  console.log(data.result.transactions)
+  const transactions = (
+    <div className="m-4">
+      <h1 className="border-b-2 text-xl font-bold">Transaction history</h1>
+      {data.result.transactions
+        .slice(0)
+        .reverse()
+        .map(
+          (transaction: {
+            periodIx: number
+            segmentIx: number
+            type: string
+            facts: { decision: boolean }
+          }) => {
+            const key = `transaction-${transaction.periodIx}-${transaction.segmentIx}`
+            return (
+              <div className="flex flex-col border-b-2 py-2" key={key}>
+                <div className="flex">
+                  <div>P {transaction.periodIx}</div>
+                  <div>Q {transaction.segmentIx}</div>
+                </div>
+                <div className="flex justify-between">
+                  <div>Decision: {transaction.type}</div>
+                  <div> {transaction.facts.decision ? 'On' : 'Off'} </div>
+                </div>
+              </div>
+            )
+          }
+        )}
+    </div>
+  )
 
+  const header = (
+    <div className="flex justify-between rounded border p-4">
+      <div className="font-bold">Game {currentGame.id}</div>
       <div className="">Current status: {currentGame.status}</div>
     </div>
   )
 
+  // TODO(JJ): Maybe we want to have two components, one for the sidebar
+  // and one for the game layout. But then the sidebar would be static
+  // -> conflict with component purpose
+  const sidebar = <div>SidebarAddons</div>
+  const gameLayout = (children: React.ReactNode) => (
```

Comment:
this is a bit of a strange way to define a component, as it is used as a function later on. makes it harder to understand that is still react (while using it through JSX notation like <GameLayout.. is much more intuitive to react devs. I would put it into a separate component (GameLayout). probably some of the data that is needed for layout and not in the children (if there is any) could also be fetched or moved to this new component, making the logic inside cockpit easier to understand